smssync/src/main/java/org/addhen/smssync/data/message/ProcessMessageResult.java

```diff
@@ -94,8 +94,8 @@ public void processMessageResult() {
             if ((response != null) && (response.isSuccess()) && (response.hasUUIDs())) {
                 final List<MessageResult> messageResults = new ArrayList<>();
                 for (String uuid : response.getUuids()) {
-                    Message message = mMessageDataSource.fetchPendingByUuid(uuid);
-                    if (message != null) {
+                    Message message = mMessageDataSource.fetchByUuid(uuid);
+                    if (message != null && !message.getStatus().equals(Message.Status.UNCONFIRMED)) {
```

Comment:
You can avoid this null check if you flip equals like this: `Message.Status.UNCONFIRMED.equals(message.getStatus())`